### ISSUE TITLE
mod_sftp sends broken response when CREATETIME attribute is requested

### ISSUE THREAD

**Issue by ppazera** [proftpd/proftpd#980]
### What I Did

In our Delphi application which uses SecureBlackBox library we are trying to download file via SFTP, and first step is to get file size. SecureBlackBox library requests CREATETIME attribute (among the others). When response is received SecureBlackBox fails to parse it because CREATETIME bit is set in response, but not enough data in the response (one timestamp is missing). Our application and secureBlackBox library works fine when using sshd as sftp server. I checked the latest mod_sftp sources and it looks like flags in response are cloned from request, CREATETIME attribute is not removed, and fxp_attrs_write does not have code section to write 'create' timestamp. Looks like RebeX library bumped into this issues, and they worked around this (https://github.com/proftpd/proftpd/issues/241#issuecomment-369892572, https://www.rebex.net/sftp.net/history.aspx#2017R6.1, "Added workaround for ProFTPd 1.3.6's mod_sftp which sends broken response when CREATETIME attribute has been requested.")

### What I Expected/Wanted

I think either CREATETIME bit should not be set in response, or create timestamp should be included in response

### ProFTPD Version and Configuration

I'm using proftpd/mod_sftp from sles15 sp1 distribution (package hub)
```
    $ proftpd -V

Compile-time Settings:
  Version: 1.3.6c (maint)
  Platform: LINUX [Linux 4.12.14-197.34-default x86_64]
  Built With:
    configure  '--host=x86_64-suse-linux-gnu' '--build=x86_64-suse-linux-gnu' '--program-prefix=' '--prefix=/usr' '--exec-prefix=/usr' '--bindir=/usr/bin' '--sbindir=/usr/sbin' '--sysconfdir=/etc' '--datadir=/usr/share' '--includedir=/usr/include' '--libdir=/usr/lib64' '--libexecdir=/usr/lib' '--localstatedir=/var' '--sharedstatedir=/var/lib' '--mandir=/usr/share/man' '--infodir=/usr/share/info' '--disable-dependency-tracking' '--disable-static' '--bindir=/usr/sbin' '--libexecdir=/usr/lib64/proftpd' '--sysconfdir=/etc/proftpd' '--localstatedir=/run/proftpd' '--enable-sendfile' '--enable-ctrls' '--enable-dso' '--enable-facl' '--enable-ipv6' '--enable-nls' '--enable-openssl' '--with-lastlog' '--with-includes=/usr/include/mysql:/usr/include/pgsql' '--with-shared=mod_auth_otp:mod_ban:mod_copy:mod_ctrls_admin:mod_deflate:mod_digest:mod_dnsbl:mod_dynmasq:mod_exec:mod_ifsession:mod_ifversion:mod_ldap:mod_load:mod_log_forensic:mod_qos:mod_quotatab:mod_quotatab_file:mod_quotatab_ldap:mod_quotatab_radius:mod_quotatab_sql:mod_radius:mod_ratio:mod_readme:mod_rewrite:mod_sftp:mod_sftp_pam:mod_sftp_sql:mod_shaper:mod_site_misc:mod_snmp:mod_sql:mod_sql_mysql:mod_sql_odbc:mod_sql_passwd:mod_sql_postgres:mod_sql_sqlite:mod_statcache:mod_tls:mod_tls_fscache:mod_tls_memcache:mod_tls_redis:mod_tls_shmcache:mod_unique_id:mod_wrap2:mod_wrap2_file:mod_wrap2_redis:mod_wrap2_sql' '--disable-ident' '--disable-strip' 'build_alias=x86_64-suse-linux-gnu' 'host_alias=x86_64-suse-linux-gnu' 'CFLAGS=-fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -D_GNU_SOURCE -DLDAP_DEPRECATED' 'CXXFLAGS=-fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -D_GNU_SOURCE -DLDAP_DEPRECATED'

  CFLAGS: -g2 -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -D_GNU_SOURCE -DLDAP_DEPRECATED -Wall -fno-omit-frame-pointer -Werror=implicit-function-declaration
  LDFLAGS: -L$(top_srcdir)/lib  -rdynamic  -L/usr/lib64 -L/usr/lib/postgresql10/lib64
  LIBS: -lacl  -lssl -lcrypto -L$(top_srcdir)/lib/libcap -lcap  -lpam -lsupp -lattr -lresolv -lresolv -lcrypt -ldl

  Files:
    Configuration File:
      /etc/proftpd/proftpd.conf
    Pid File:
      /run/proftpd/proftpd.pid
    Scoreboard File:
      /run/proftpd/proftpd.scoreboard
    Header Directory:
      /usr/include/proftpd
    Shared Module Directory:
      /usr/lib64/proftpd

  Info:
    + Max supported UID: 4294967295
    + Max supported GID: 4294967295

  Features:
    - Autoshadow support
    + Controls support
    - curses support
    - Developer support
    + DSO support
    + IPv6 support
    + Largefile support
    + Lastlog support
    - Memcache support
    + ncursesw support
    + NLS support
    - Redis support
    - Sodium support
    + OpenSSL support (FIPS enabled)
    - PCRE support
    + POSIX ACL support
    + Shadow file support
    + Sendfile support
    + Trace support
    + xattr support

  Tunable Options:
    PR_TUNABLE_BUFFER_SIZE = 1024
    PR_TUNABLE_DEFAULT_RCVBUFSZ = 8192
    PR_TUNABLE_DEFAULT_SNDBUFSZ = 8192
    PR_TUNABLE_ENV_MAX = 2048
    PR_TUNABLE_GLOBBING_MAX_MATCHES = 100000
    PR_TUNABLE_GLOBBING_MAX_RECURSION = 8
    PR_TUNABLE_HASH_TABLE_SIZE = 40
    PR_TUNABLE_LOGIN_MAX = 256
    PR_TUNABLE_NEW_POOL_SIZE = 512
    PR_TUNABLE_PATH_MAX = 4096
    PR_TUNABLE_SCOREBOARD_BUFFER_SIZE = 80
    PR_TUNABLE_SCOREBOARD_SCRUB_TIMER = 30
    PR_TUNABLE_SELECT_TIMEOUT = 30
    PR_TUNABLE_TIMEOUTIDENT = 10
    PR_TUNABLE_TIMEOUTIDLE = 600
    PR_TUNABLE_TIMEOUTLINGER = 10
    PR_TUNABLE_TIMEOUTLOGIN = 300
    PR_TUNABLE_TIMEOUTNOXFER = 300
    PR_TUNABLE_TIMEOUTSTALLED = 3600
    PR_TUNABLE_XFER_SCOREBOARD_UPDATES = 10
```

In addition, we need to see *all* of the ProFTPD configuration files you are
using (*minus* any sensitive information like passwords, of course).  Armed
with the version and configuration data, then, we can set up ProFTPD locally
using the same configuration, and see what happens.

**Comment by Castaglia** [comment]
@ppazera can you verify that https://github.com/proftpd/proftpd/commit/1708a282efbb3ad333361e5965586c0a717020be addresses the issue for you?

**Comment by ppazera** [comment]
@Castaglia yes, it works now, thanks for a quick fix!

**Comment by ppazera** [comment]
@Castaglia one kind of unrelated question. After upgrading proftpd to latest sources SexureBlackBox library stopped working because it does not recognize EXT_INFO (7) packets and disconnects. I found NoExtensionNegotiation mod_sftp option and it helped, but that doesn't sound like correct solution because other clients may want to use extension negotiation. Maybe server should not send EXT_INFO when client does not signal "ext-info-c"? (I did not check RFC closely, found only draft)
